envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/GroupChangeWatcher.kt

```diff
@@ -38,12 +38,15 @@ internal class GroupChangeWatcher(
         val groups = cache.groups()
         metrics.setCacheGroupsCount(groups.size)
         if (oldGroups != groups) {
-            emitNewGroupsEvent()
+            val d1 = (groups - oldGroups)
+            val d2 = (oldGroups - groups)
+            val symmetricDifference = d1.union(d2)
```

Comment:
I wonder if `Sets.symmetricDifference(oldGroups.toSet(), groups.toSet())` wouldn't be more effective. From the "minus" operator docs:
> The [elements] collection may be converted to a [HashSet] to speed up the operation
So it will be converted to set anyway...